javascript/packages/orchestrator/src/spawner.ts

```diff
@@ -0,0 +1,249 @@
+import { Node, NodeMultiAddress, Parachain, fileMap } from "./types";
+const debug = require("debug")("zombie::spawner");
+
+export const spawnNode = async (
+  client: Client,
+  node: Node,
+  network: Network,
+  bootnodes: string[],
+  filesToCopy: fileMap[],
+  opts: {
+    silent: boolean;
+    inCI: boolean;
+    monitorIsAvailable: boolean;
+    userDefinedTypes?: any;
+    local_ip?: string;
+    jaegerUrl?: string;
+  },
+  parachain?: Parachain,
+): Promise<NodeMultiAddress> => {
+  console.log("parachain", parachain);
+  const namespace = client.namespace;
+  const { genBootnodeDef, genNodeDef, replaceNetworkRef } = getProvider(
+    client.providerName,
+  );
+
+  let parachainSpecId;
+  // for relay chain we can have more than one bootnode.
+  if (node.zombieRole === "node" || node.zombieRole === "collator")
+    node.bootnodes = node.bootnodes.concat(bootnodes);
+
+  if (opts.jaegerUrl) node.jaegerUrl = opts.jaegerUrl;
+
+  debug(`creating node: ${node.name}`);
+  const podDef = await (node.name === "bootnode"
+    ? genBootnodeDef(namespace, node)
+    : genNodeDef(namespace, node));
+
+  let finalFilesToCopyToNode = [...filesToCopy];
+
+  // add spec file if is provided
+  if (parachain?.specPath) {
+    finalFilesToCopyToNode.push({
+      localFilePath: parachain.specPath,
+      remoteFilePath: `${client.remoteDir}/${node.chain}-${parachain.id}.json`,
+    });
+    parachainSpecId = await getChainIdFromSpec(parachain.specPath);
+  }
+  for (const override of node.overrides) {
+    finalFilesToCopyToNode.push({
+      localFilePath: override.local_path,
+      remoteFilePath: `${client.remoteDir}/${override.remote_name}`,
+    });
+  }
+
+  let keystoreLocalDir;
+  if (node.accounts) {
+    // check if the node directory exists if not create (e.g for k8s provider)
+    let nodeFilesPath = client.tmpDir;
+    if (parachain && parachain.name) nodeFilesPath += `/${parachain.name}`;
+    nodeFilesPath += `/${node.name}`;
+
+    await makeDir(nodeFilesPath, true);
+
+    const isStatemint = parachain && parachain.chain?.includes("statemint");
+    const keystoreFiles = await generateKeystoreFiles(
+      node,
+      nodeFilesPath,
+      isStatemint,
+    );
+    keystoreLocalDir = path.dirname(keystoreFiles[0]);
+  }
+
+  // replace all network references in command
+  replaceNetworkRef(podDef, network);
+
+  await client.spawnFromDef(
+    podDef,
+    finalFilesToCopyToNode,
+    keystoreLocalDir,
+    parachainSpecId || network.chainId,
+    node.dbSnapshot,
+  );
+
+  const [nodeIp, nodePort] = await client.getNodeInfo(podDef.metadata.name);
+  const nodeMultiAddress = await generateNodeMultiAddress(
+    node.key!,
+    node.args,
+    nodeIp,
+    nodePort,
+    true,
+    node.p2pCertHash,
+  );
+
+  // multiAddressByNode[podDef.metadata.name] = nodeMultiAddress;
+
+  // if (node.addToBootnodes) {
+  //   bootnodes.push(nodeMultiAddress);
+  //   await addBootNodes(chainSpecFullPath, bootnodes);
+  // }
```

Comment:
remove comments

Reply:
Thanks, I think also I have to revisit the way bootnodes are handled.